From c0fb4855943231125a11cf60da1840e7aa0c9771 Mon Sep 17 00:00:00 2001
From: Brian Harring <ferringb@chromium.org>
Date: Fri, 23 Dec 2011 18:16:15 -0800
Subject: [PATCH] CHROMIUM: make-3.82: hack to fix differing behaviour between
 3.81 and 3.82

BUG=chromium-os:19882
TEST=emerge make-3.82 via I7d0a6e25, emerge-x86-alex chromeos-kernel

Change-Id: I6accabc10e93fe12debba42667900e4e0463cb41
Signed-off-by: Brian Harring <ferringb@chromium.org>
Reviewed-on: https://gerrit.chromium.org/gerrit/13474
Reviewed-by: Olof Johansson <olofj@chromium.org>
(cherry picked from commit aac3e80b6ca8b49036786edfb5ff0aead2426138)
---
 scripts/Makefile.fwinst | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/scripts/Makefile.fwinst b/scripts/Makefile.fwinst
index 6bf8e87..689204f 100644
--- a/scripts/Makefile.fwinst
+++ b/scripts/Makefile.fwinst
@@ -27,7 +27,7 @@ endif
 installed-mod-fw := $(addprefix $(INSTALL_FW_PATH)/,$(mod-fw))
 
 installed-fw := $(addprefix $(INSTALL_FW_PATH)/,$(fw-shipped-all))
-installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)/.
+installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)
 
 # Workaround for make < 3.81, where .SECONDEXPANSION doesn't work.
 PHONY += $(INSTALL_FW_PATH)/$$(%) install-all-dirs
@@ -36,6 +36,14 @@ $(INSTALL_FW_PATH)/$$(%): install-all-dirs
 install-all-dirs: $(installed-fw-dirs)
 	@true
 
+
+# hack for make-3.82 support; without this (admittedly daft) rule,
+# firmware installation fails since it can't find a matching
+# rule for $(INSTALL_FW_PATH/..
+# leave the hack in place till the root cause is fully found, without
+# it make-3.82 cannot be used.
+$(INSTALL_FW_PATH)/./: $(INSTALL_FW_PATH)
+
 quiet_cmd_install = INSTALL $(subst $(srctree)/,,$@)
       cmd_install = $(INSTALL) -m0644 $< $@
 
